src/main/java/org/sonatype/repository/helm/HelmUploadHandler.java

```diff
@@ -0,0 +1,107 @@
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.rest.UploadDefinitionExtension;
+import org.sonatype.nexus.repository.security.ContentPermissionChecker;
+import org.sonatype.nexus.repository.security.VariableResolverAdapter;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreBlob;
+import org.sonatype.nexus.repository.upload.ComponentUpload;
+import org.sonatype.nexus.repository.upload.UploadDefinition;
+import org.sonatype.nexus.repository.upload.UploadHandlerSupport;
+import org.sonatype.nexus.repository.upload.UploadResponse;
+import org.sonatype.nexus.repository.view.PartPayload;
+import org.sonatype.nexus.transaction.Transaction;
+import org.sonatype.nexus.transaction.UnitOfWork;
+import org.sonatype.repository.helm.internal.AssetKind;
+import org.sonatype.repository.helm.internal.HelmFormat;
+import org.sonatype.repository.helm.internal.hosted.HelmHostedFacet;
+import org.sonatype.repository.helm.internal.metadata.HelmAttributes;
+import org.sonatype.repository.helm.internal.util.HelmAttributeParser;
+import org.sonatype.repository.helm.internal.util.HelmPathUtils;
+import org.sonatype.repository.helm.internal.util.TgzParser;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Set;
+
+import static org.sonatype.repository.helm.internal.util.HelmDataAccess.HASH_ALGORITHMS;
+
+/**
+ * Support helm upload for web page
+ * @author yinlongfei
+ */
+@Singleton
+@Named(HelmFormat.NAME)
+public class HelmUploadHandler extends UploadHandlerSupport {
+    private UploadDefinition definition;
+    private final ContentPermissionChecker contentPermissionChecker;
+    private final VariableResolverAdapter variableResolverAdapter;
+
+    private final HelmAttributeParser helmPackageParser;
+
+
+    @Inject
+    public HelmUploadHandler(final ContentPermissionChecker contentPermissionChecker,final HelmAttributeParser helmPackageParser, @Named("simple") final VariableResolverAdapter variableResolverAdapter, final Set<UploadDefinitionExtension> uploadDefinitionExtensions){
+        super(uploadDefinitionExtensions);
+        this.contentPermissionChecker = contentPermissionChecker;
+        this.variableResolverAdapter = variableResolverAdapter;
+        this.helmPackageParser = helmPackageParser;
+    }
+
+
+    @Override
+    public UploadResponse handle(Repository repository, ComponentUpload upload) throws IOException {
+        HelmHostedFacet facet = repository.facet(HelmHostedFacet.class);
+        StorageFacet storageFacet = repository.facet(StorageFacet.class);
+
+
+        return TransactionalStoreBlob.operation.withDb(storageFacet.txSupplier()).throwing(IOException.class).call(()->{
```

Comment:
I think this is following my bad pattern, the temp blob creation and extraction of metadata ought to occur outside of the transaction - e.g. https://github.com/sonatype/nexus-public/blob/master/plugins/nexus-repository-pypi/src/main/java/org/sonatype/nexus/repository/pypi/PyPiUploadHandler.java#L76